google-analytics-date-selector.html

```diff
@@ -55,27 +50,68 @@
       <input
         id="endDate"
         type="date"
-        value="{{endDate}}"
+        value="{{endDate::change}}"
         min="{{startDate}}"
         max="{{maxEndDate}}">
     </span>
   </template>
-  <script>
-
-    (function() {
-      /**
-       * Fired when the users changes the start or end date.
-       *
-       * @param {Object} query The updated query params.
-       * @event analytics-dashboard-control-change
-       */
-
-
-      var nDaysAgo = /(\d+)daysAgo/;
-      var dateFormat = /\d{4}\-\d{2}\-\d{2}/;
-
-      Polymer('google-analytics-date-selector', {
-
+</dom-module>
+<script>
+
+  (function() {
+    /**
+     * Fired when the users changes the start or end date.
+     *
+     * @param {Object} query The updated query params.
+     * @event analytics-dashboard-control-change
+     */
+
+
+    var nDaysAgo = /(\d+)daysAgo/;
+    var dateFormat = /\d{4}\-\d{2}\-\d{2}/;
+
+    /**
+     * Convert a date acceptable to the Core Reporting API (e.g. `today`,
+     * `yesterday` or `NdaysAgo`) into the format YYYY-MM-DD. Dates
+     * already in that format are simply returned.
+     * @return {string} The formatted date.
+     */
+    function convertDate(str) {
```

Comment:
done a while ago, not sure. I usually bunch up all loose code either on top, or bottom.